protocol/daemons/bridge/client/client.go

```diff
@@ -73,140 +88,59 @@ func Start(
 	// Initialize an Ethereum client from an RPC endpoint.
 	ethClient, err := ethclient.Dial(flags.Bridge.EthRpcEndpoint)
 	if err != nil {
-		logger.Error("Failed to establish connection to Ethereum node", "error", err)
+		c.logger.Error("Failed to establish connection to Ethereum node", "error", err)
 		return err
 	}
 	defer func() { ethClient.Close() }()
 
-	// Run the main task loop at an interval.
 	ticker := time.NewTicker(time.Duration(flags.Bridge.LoopDelayMs) * time.Millisecond)
-	for ; true; <-ticker.C {
-		if err := RunBridgeDaemonTaskLoop(
-			ctx,
-			logger,
-			ethClient,
-			queryClient,
-			serviceClient,
-		); err != nil {
-			// TODO(DEC-947): Move daemon shutdown to application.
-			logger.Error("Bridge daemon returned error", "error", err)
-		}
-	}
+	stop := make(chan bool, 1)
+	// Run the main task loop at an interval.
+	StartBridgeDaemonTaskLoop(
+		ctx,
+		c,
+		ticker,
+		stop,
+		&SubTaskRunnerImpl{},
+		ethClient,
+		queryClient,
+		serviceClient,
+	)
 
 	return nil
 }
 
-// RunBridgeDaemonTaskLoop does the following:
-// 1) Fetches configuration information by querying the gRPC server.
-// 2) Fetches Ethereum events from a configured Ethereum client.
-// 3) Sends newly-recognized bridge events to the gRPC server.
-func RunBridgeDaemonTaskLoop(
+// StartBridgeDaemonTaskLoop operates the continuous loop that runs the bridge daemon. It receives as arguments
+// a ticker and a stop channel that are used to control and halt the loop.
+func StartBridgeDaemonTaskLoop(
```

Comment:
`StartBridgeDaemonTaskLoop` takes a ticker and a stop channel, similar to how go subroutines for price daemon are designed, and uses the stop channel to abort the go routien. This is esp. useful for tests!